sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/ShuffledBatchRDD.scala

```diff
@@ -175,22 +178,25 @@ class ShuffledBatchRDD(
         val partitionSize = blocksByAddress.flatMap(_._2).map(_._2).sum
         (reader, partitionSize)
 
-      case PartialReducerPartitionSpec(reducerIndex, startMapIndex, endMapIndex) =>
-        val reader = shuffleManagerShims.getReader(
-          SparkEnv.get.shuffleManager,
-          dependency.shuffleHandle,
-          startMapIndex,
-          endMapIndex,
-          reducerIndex,
-          reducerIndex + 1,
-          context,
-          sqlMetricsReporter)
-        val blocksByAddress = shim.getMapSizesByExecutorId(
-          dependency.shuffleHandle.shuffleId, 0, Int.MaxValue, reducerIndex, reducerIndex + 1)
-        val partitionSize = blocksByAddress.flatMap(_._2)
-            .filter(tuple => tuple._3 >= startMapIndex && tuple._3 < endMapIndex)
-            .map(_._2).sum
-        (reader, partitionSize)
+      case prps: PartialReducerPartitionSpec =>
+        ShimLoader.getSparkShims.getShuffleManagerShims.toGpu(prps) match {
```

Comment:
Seems like we could avoid shimming this entirely (at least for now) by avoiding the `unapply` on `ParitalReducerPartitionSpec`.  Instead we could reference `prps.startMapIndex` etc. in the use below.